cpp/include/cudf/copying.hpp

```diff
@@ -239,28 +242,34 @@ table scatter(std::vector<gdf_scalar> const& source,
  * The number of elements in the gather_map must equal the number of rows in the
  * destination columns.
  *
- * If any index in the gather_map is outside the range [0, num rows in
- * source_columns), the result is undefined.
- *
- * If the same index appears more than once in gather_map, the result is
- * undefined.
+ * If any index in the gather_map is outside the range `[-n, n)`, where
+ * `n` is the number of rows in the source column, the behavior is underfined.
+ * Optionally, a bounds check can be performed and an error can be raised.
  *
  * @param[in] source_table The input columns whose rows will be gathered
- * @param[in] gather_map An array of indices that maps the rows in the source
+ * @param[in] gather_map A column containing indices that maps the rows in the source
  * columns to rows in the destination columns.
  * @param[out] destination_table A preallocated set of columns with a number
  * of rows equal in size to the number of elements in the gather_map that will
  * contain the rearrangement of the source columns based on the mapping. Can be
  * the same as `source_table` (in-place gather).
  *
  */
-void gather(table const* source_table, gdf_index_type const gather_map[],
+void gather(table const* source_table, gdf_column const& gather_map,
 	    table* destination_table, bool check_bounds=false);
 
-void gather(table const* source_table, gdf_column const gather_map,
+/**
+ * @overloads
```

Comment:
Hmmm, @overload doesn't work that way. You have to give the function signature of the function this one overloads. You should also add additional documentation if the behavior is different for the overload.

http://www.doxygen.nl/manual/commands.html#cmdoverload

Reply:
Gotcha. My impression was that all `@overload` does is include the message `This is an overloaded member function, provided for convenience. It differs from the above function only in what argument(s) it accepts.`

Reply:
Actually I think that's really all `@overload` does. Adding the function signature following `@overload` made it disappear from the documentation altogether (I tried both the signature of the overloading function as well as the overloaded function - neither worked). I also tried toggling the `SORT_MEMBER_DOCS` option in `Doxyfile` to see if it had any effect, but no.

Reply:
Done